source/includes/read/distinct.kt

```diff
@@ -0,0 +1,61 @@
+import com.mongodb.ConnectionString
+import com.mongodb.MongoClientSettings
+import com.mongodb.client.model.Filters.and
+import com.mongodb.client.model.Filters.eq
+import com.mongodb.kotlin.client.MongoClient
+
+// start-data-class
+data class Restaurant(
+    val name: String,
+    val borough: String,
+    val cuisine: String
+)
+// end-data-class
+
+fun main() {
+    val uri = "<connection string URI>"
+
+    val settings = MongoClientSettings.builder()
+        .applyConnectionString(ConnectionString(uri))
+        .retryWrites(true)
+        .build()
+
+    val mongoClient = MongoClient.create(settings)
+    val database = mongoClient.getDatabase("sample_restaurants")
+    val collection = database.getCollection<Restaurant>("restaurants")
+
+    // start-distinct
+    val results = collection.distinct<String>(Restaurant::borough.name)
+
+    results.forEach { result ->
```

Comment:
`forEach` appears to be deprecated according to the kotlin doc, which now recommends using `collect` instead. I'm unsure if the usage of `forEach` is specific to this PR or if we also need to review previously merged documentation for consistency. Could we verify if updates are needed elsewhere?

Reply:
Frankly, a lot of previously reviewed documentation in this repo has been using `forEach()`.

Reply:
Thank you for clarifying, and my apologies for the confusion. You're correct; I mixed up the contexts between coroutines and synchronous implementations. `forEach` is indeed not deprecated for synchronous use in Kotlin, and collect is recommended specifically for asynchronous streams in coroutines.